UI/FleetWnd.cpp

```diff
@@ -937,16 +925,8 @@ namespace {
         // client height should never be less than the height of the space resereved for the icon
         if (m_ship_icon)
             m_ship_icon->Resize(GG::Pt(DataPanelIconSpace().x, ClientHeight()));
-        if (m_scrap_indicator)
-            m_scrap_indicator->Resize(GG::Pt(DataPanelIconSpace().x, ClientHeight()));
-        if (m_colonize_indicator)
-            m_colonize_indicator->Resize(GG::Pt(DataPanelIconSpace().x, ClientHeight()));
-        if (m_invade_indicator)
-            m_invade_indicator->Resize(GG::Pt(DataPanelIconSpace().x, ClientHeight()));
-        if (m_bombard_indicator)
-            m_bombard_indicator->Resize(GG::Pt(DataPanelIconSpace().x, ClientHeight()));
-        if (m_scanline_control)
-            m_scanline_control->Resize(GG::Pt(DataPanelIconSpace().x, ClientHeight()));
+        if (m_ship_icon_overlay)
```

Comment:
this only supports one overlay per ship, but ships can have more than one of these statuses at the same time.